keras/src/layers/preprocessing/image_preprocessing/random_flip.py

```diff
@@ -93,8 +96,45 @@ def transform_bounding_boxes(
         bounding_boxes,
         transformation,
         training=True,
+        input_shape=None,
     ):
-        raise NotImplementedError
+        flips = transformation["flips"][0]
+        input_height, input_width = input_shape
+
+        bounding_boxes = convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            height=input_height,
+            width=input_width,
+        )
+
+        x1, y1, x2, y2 = self.backend.numpy.split(
```

Comment:
You can process all coordinates at once.

Additionally, you will need to clip the bboxes at the end since the new coordinates might fall outside the image.

See this reference implementation: https://github.com/keras-team/keras-cv/blob/master/keras_cv/src/layers/preprocessing/random_flip.py#L157-L184

Reply:
@fchollet 
I have a question regarding batch processing.

I believe that applying batch processing to transform_bounding_boxes could significantly improve performance. However, it seems that keras.utils.bounding_boxes.convert_format is currently implemented to handle only a single image at a time.

I tried using this method for batch processing, but I’m wondering—do you think it would be necessary to modify this method to support batch processing before proceeding with this logic?

The reason I’m asking is that each image in my test dataset has a different size.

Reply:
> I tried using this method for batch processing, but I’m wondering—do you think it would be necessary to modify this method to support batch processing before proceeding with this logic?

Yes, `convert_format` should work with batches, I'm surprised it doesn't do that currently

> The reason I’m asking is that each image in my test dataset has a different size.

They need to be standardized to a shared size by the time they hit the layer.

Reply:
First of all, I apologize for my limited knowledge in computer vision. I'm not completely sure I understand the convert_format method in keras_cv and keras.

The documentation mentions that the method expects the bounding box input in the format [batch_size, num_boxes, 4]. However, I noticed that the number of bounding boxes (num_boxes) varies between images in my test set.

Does this mean I need to preprocess the bounding boxes to ensure they have the same number per image before using convert_format? Or does this method support images with different numbers of bounding boxes as well? Thank you in advance for any clarification.

Reply:
> However, I noticed that the number of bounding boxes (num_boxes) varies between images in my test set.

Right, bounding boxes need to be padded. You pick a max number of bounding boxes: if you have more, you drop the remainder, and if you have less, you pad to the max number using a padding value such as -1.

Reply:
Note that the padding can be done on a per-batch basis rather than globally. The only goal is that layers should process dense tensors.

Reply:
Thank you for the detailed explanation. I’ve updated my code according to your guidance.